hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java

```diff
@@ -64,6 +64,9 @@ public InternalSerializationService createSerializationService(byte version) {
         try {
             ClientConfig config = client.getClientConfig();
             ClassLoader configClassLoader = config.getClassLoader();
+            if (null == configClassLoader) {
```

Comment:
Using our classLoaderUtil for this seems like correct solution
`klass = ClassLoaderUtil.loadClass(in.getClassLoader(), klassName);`